### PR TITLE
doxygen: documentation hierarchy fixes

### LINKED ISSUE #3391: doc: wrong hierarchy level for some modules
For some weird reasons some modules (e.g. 6LoWPAN or IEEE802.15.4e) show up at the wrong hierarchy level at http://doc.riot-os.org/modules.html. Building the documentation locally seems to generate the expected hierarchy. The documentation on the web page is generated by doxygen version 1.8.8 - locally I have 1.8.9.1. Can someone cross check?

### PR DESCRIPTION
**Pull request by OlegHahm**
fixes #3391

### CODE REVIEW

File: cpu/sam3/doc.txt [hunk 1]
```diff
@@ -6,5 +6,5 @@
 
 /**
  * @defgroup        cpu_sam3x8e_definitions Atmel SAM3X8E Definitions
```
Comment (gebart) [inline cpu/sam3/doc.txt]: could you update all the group names to `cpu_sam3_xxxx`?

Comment (OlegHahm) [inline cpu/sam3/doc.txt]: will do

**Comment by OlegHahm** [comment]
addressed comment

**Comment by haukepetersen** [comment]
ACK when Travis....

**Pull request merged**